### ISSUE TITLE
Intellij Project root being marked as exluded

### ISSUE THREAD

**Issue by Extreme66** [Camelcade/Perl5-IDEA#2030]
Intellij Ultimate 2019.1
Build #IU-191.6183.87

Perl Plugin v2019.1.1

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

  * Centos 7

I am having an issue where when I enable the perl plugin it is marking my project root as excluded for non-perl projects. I have tested and it seems to be this plugin, it didn't happen in 2018. versions of Intellij, but once I switched I am noticing it. I am on a PHP project and tested all of the other plugins as well, this is the only one im seeing this happen with.

From what i reported to jetbrains...

Hi, I am checking out a fresh project from git and all of my files are being marked as excluded once indexing completes. I cannot un-exclude them for some reason and they are not showing up as excluded on the right hand pane.
![Screenshot](https://user-images.githubusercontent.com/3781673/56389787-725d1f80-61f8-11e9-98f2-da0b955b3533.png)
 I also added the debug log from idea if that helps

we were able to determine that something with the Perl5 plugin. This is what we did. We disabled the Perl5 plugin, flushed the caches and restarted. All looked good, no more excluded files. We enabled (a few times) the Perl5 plugin and the issue re-appeared. Although there are no perl files in this (PHP) project. Near the end of the rescan when we restarted it would mark the files being excluded and right at that time these lines are from the idea.log. Disabling the Perl5 plugin makes the problem go away. Where should i go from here. Hopefully this helps.

```
2019-04-17 12:22:13,735 [ 189247] INFO - ServersWorkspaceStateComponent - Replacing existing server with the shared one with the same id 
2019-04-17 12:22:15,143 [ 190655] INFO - pl.ProjectRootManagerComponent - project roots have changed 
2019-04-17 12:22:15,549 [ 191061] INFO - .diagnostic.PerformanceWatcher - Pushing properties took 256ms; general responsiveness: ok; EDT responsiveness: ok 
2019-04-17 12:22:16,567 [ 192079] INFO - .diagnostic.PerformanceWatcher - Indexable file iteration took 991ms; general responsiveness: ok; EDT responsiveness: ok 
2019-04-17 12:22:16,570 [ 192082] INFO - indexing.UnindexedFilesUpdater - Unindexed files update started: 6308 files to update 
2019-04-17 12:22:18,602 [ 194114] WARN - ConfigurableExtensionPointUtil - ignore deprecated groupId: language for id: preferences.language.Kotlin.scripting 
2019-04-17 12:22:20,290 [ 195802] INFO - j.ide.plugins.RepositoryHelper - using cached plugin list (updated at 4/17/19 12:16 PM) 
2019-04-17 12:22:28,587 [ 204099] WARN - com.intellij.util.xmlb.Binding - no accessors for class com.perl5.lang.perl.idea.sdk.host.local.PerlLocalHostData 
2019-04-17 12:22:28,587 [ 204099] WARN - com.intellij.util.xmlb.Binding - no accessors for class com.perl5.lang.perl.idea.sdk.implementation.porters.PerlPortersData 
2019-04-17 12:22:28,587 [ 204099] WARN - com.intellij.util.xmlb.Binding - no accessors for class com.perl5.lang.perl.idea.sdk.versionManager.system.PerlSystemData 
2019-04-17 12:22:28,611 [ 204123] INFO - rationStore.ComponentStoreImpl - Saving appFileTypeManager took 28 ms 
2019-04-17 12:22:28,761 [ 204273] INFO - rationStore.ComponentStoreImpl - Saving Module: 'infocenter-swarm-stacks-rpm'ExternalFacetManager took 14 ms 
2019-04-17 12:22:28,883 [ 204395] INFO - rationStore.ComponentStoreImpl - Saving Project '/home/jacy.ciszkowski/projects/infocenter/infocenter-3.46' infocenter-3.46CommitMessageInspectionProfile took 23 ms 
2019-04-17 12:22:28,890 [ 204402] INFO - mponents.impl.stores.StoreUtil - saveProjectsAndApp took 487 ms
```

**Comment by hurricup** [comment]
1. There is no some kind of special perl project with special logic. You may create a perl module, but it's more like legacy thing and perl works in parallel for any project. 
2. As far as I remember there is no any logic related to excluding dirs in perl5 plugin. 
3. I can't figure out anything from this log.
4. Does this happen on some empty and/or other project?
5. Does the same happen on other machine (if possible to check)? 
6. Does this happen with only perl plugin enabled? Btw, would be nice to have a startup part of your logs. 
7. Probably worth to be reported to the jetbrains youtrack.

**Comment by hurricup** [comment]
Any updates?

**Comment by Extreme66** [comment]
Hello, I have reported it to jetbrains, so far i havent had any progress i didnt see this happen in 2018.x only 2019.x Im still trying to isolate what is causing it.

**Comment by hurricup** [comment]
And what about other questions I've asked?

**Comment by hurricup** [comment]
Also, could you provide a link to the JB issue?

**Comment by Extreme66** [comment]
Of course, sorry about that.

There is no some kind of special perl project with special logic. You may create a perl module, but it's more like legacy thing and perl works in parallel for any project.
As far as I remember there is no any logic related to excluding dirs in perl5 plugin.
I can't figure out anything from this log.
Does this happen on some empty and/or other project? I don't see it on my perl projects, and only have 1 other php project.
Does the same happen on other machine (if possible to check)? Yes i was able to duplicate with the same project on 3 machines as well.
Does this happen with only perl plugin enabled? Btw, would be nice to have a startup part of your logs.
Probably worth to be reported to the jetbrains youtrack.
https://intellij-support.jetbrains.com/hc/en-us/requests/2059933
# 2059933

**Comment by hurricup** [comment]
Oh, i thought it was a youtrack issue. Is your project open source?
